### PR TITLE
FIX: macOS compatibility

### PR DESCRIPTION
**Pull request by reneme**
On macOS `ncursesw` is not available. Instead one needs to use `ncurses` (without `...w`) and define `-DNCURSES_WIDECHAR`. This patch for the meson build system should allow compiling and running on macOS.

Note that `ncurses` needs to be installed on the system first:

```bash
brew install ncurses
export PKG_CONFIG_PATH="/usr/local/opt/ncurses/lib/pkgconfig"
```

Screenshot 2019-09-23 at 22 20 52

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -10,7 +10,14 @@ project('icecream-sundae', 'cpp',
...
 glib = dependency('glib-2.0')
-ncurses = dependency('ncursesw')
+
+if (build_machine.system() == 'linux')
+    ncurses = dependency('ncursesw')
+    ncurses_cxxflag = ''
+else
+    ncurses = dependency('ncurses')
+    ncurses_cxxflag = '-DNCURSES_WIDECHAR'
+endif
```
Comment (JPEWdev) [inline meson.build]: Could we do this more dynamically:

```
ncurses = dependency('ncursesw', required: false)
ncurses_cxxflags = ''
if not ncurses.found()
    ncurses = dependency('ncurses')
    ncurses_cxxflag = '-DNCURSES_WIDECHAR'
endif
```

File: meson.build [hunk 2]
```diff
@@ -33,8 +40,11 @@ int main(int argc, char **argv)
 if not cxx.links(code, dependencies: deps, name: 'libicecc links dynamically')
     icecc = dependency('icecc', static: true)
     liblzo2 = cxx.find_library('lzo2')
-    libcapng = dependency('libcap-ng')
-    deps += [libcapng, liblzo2]
+    deps += [liblzo2]
+    if build_machine.system() == 'linux'
+        libcapng = dependency('libcap-ng')
+        deps += [liblzo2]
```
Comment (JPEWdev) [inline meson.build]: Likewise:

```
libcapng = depedency('libcap-ng', required: false)
if libcapng.found()
    deps += [libcapng]
endif

File: meson.build (hunk 1 above)
Comment (reneme) [reply meson.build]: I simplified that even further to 

```meson
ncurses = dependency('ncursesw')
ncurses_cxxflag = '-DNCURSES_WIDECHAR'
```

after realizing that the `pkgconfig` provided by `brew` on macOS actually has a configuration for `ncursesw`. Unfortunately the `NCURSES_WIDECHAR` definition still needs to be provided.

File: meson.build (hunk 2 above)
Comment (reneme) [reply meson.build]: Apparently `meson` can actually deal with optional libraries in the `deps` list even if they are not found. Hence, the `if libcapng.found()` doesn't seem to be necessary.

File: meson.build [hunk 3]
```diff
@@ -33,8 +35,9 @@ int main(int argc, char **argv)
 if not cxx.links(code, dependencies: deps, name: 'libicecc links dynamically')
     icecc = dependency('icecc', static: true)
     liblzo2 = cxx.find_library('lzo2')
-    libcapng = dependency('libcap-ng')
-    deps += [libcapng, liblzo2]
+    libcapng = dependency('libcap-ng', required: false)
+    libzstd = cxx.find_library('zstd', required: false)
```
Comment (reneme) [inline meson.build:39]: On my work machine I needed to link against `zstd` because it was required by `libicecc.a`. Was that added to `icecc` recently?

Comment (JPEWdev) [reply meson.build:39]: Yes, it was added in the last release

**Pull request merged**